Added content for cloud auto migration (#3859)
temporalio/documentation: docs/production-deployment/cloud/migrate/automated.mdx

   the _[AbortMigrationRequest](https://pkg.go.dev/github.com/temporalio/tcld#readme-abort-a-migration)_ to
   roll-back changes without impacting your workflows. These APIs provide granular control over every step of the process, ensuring transparency and flexibility.

### Current limitations

The following are known limitations for the Pre-release phase of this service.

- OSS server versions 1.22 or newer are required. Refer to the [upgrade](https://docs.temporal.io/self-hosted-guide/upgrade-server#upgrade-server) procedure as needed.
- History shard counts must be a multiple of two.
- Enabling payload encryption as part of migration is not yet supported. If payloads are already [encrypted](https://docs.temporal.io/payload-codec#encryption) in your self-hosted server
  via data converter, then they will remain encrypted during and after migration.
- If you are using multi-cluster replication in your self-hosted setup and have previously failed over namespaces, then this may impact your eligibility for automated migration.

> - If you have multiple self-hosted servers and they are all configured with the same cluster name (by default Temporal uses active as cluster name), they cannot be connected to

Reviewer: Question here: by default Temporal uses active as cluster name. Is "active" the cluster name?

Author: Yes, according to the source eng doc.

- OSS supports cross-namespace commands (e.g., parent-child, SignalExternal, CancelExternal) through the _system.enableCrossNamespaceCommands_ configuration. This
  configuration is disabled on Temporal Cloud. If cross-namespace calls exist within workflow code, they must be updated or removed prior to migration.

### Getting started

In preparing for migration, you must first provide qualification details to Temporal via a support ticket. If eligible, the Temporal team will work with you to facilitate your migration.

Submit a support ticket with the following details:

- A list of your Temporal accounts
- Target Temporal Cloud service regions.
- For each cluster, provide the server configuration by running `tctl --address <frontend:7233> admin cluster describe`
- Metrics for the namespaces to be migrated
  - number of open/closed workflows
  - storage used
  - retention policy
  - any custom search attributes
  - peak request per second (RPS) and action per second (APS)
- If you use SQL-based datastore for visibility, and you use custom search attributes, provide _CustomSearchAttributeAliases_ of your namespace by running

>   temporal operator namespace describe using the [latest temporal cli](https://github.com/temporalio/cli).

Reviewer: ```suggestion
  temporal operator namespace describe using the [latest Temporal CLI](https://github.com/temporalio/cli).
```

:::warning

Proceed only when your request has been approved by Temporal.

:::

### Create cloud-side resources

Cloud-side resources must be in-place prior to starting a migration. Complete the following procedure.

1. Create one or more empty namespaces in Temporal Cloud to serve as the migration targets. Since migration cannot proceed into a namespace that's already in use, these namespaces should
   remain empty (no workflows).
2. Create a support ticket requesting these namespaces be configured with [system limits](https://docs.temporal.io/cloud/limits)
   (including APS/RPS) matching your existing self-hosted workload.

> 3. Verify that migration endpoint has been created in Temporal Cloud (e.g., `your-endpoint.{your-acct}.tmprl.cloud`). If you don't have one, request one via a support ticket.

Reviewer: ```suggestion
3. Verify that a migration endpoint has been created in Temporal Cloud (e.g., `your-endpoint.{your-acct}.tmprl.cloud`). If you don't have one, request one via a support ticket.
```

   clusterInformation:
      _NO_CHANGE_:
         enabled: true
         initialFailoverVersion: [1,100] # pick a unique number between 1 and 100 for each server 
         rpcName: _NO_CHANGE_
         rpcAddress: _NO_CHANGE_
   ```

3. Run `temporal operator cluster describe` to check the output. The following output is expected:

   ```yaml
   "failoverVersionIncrement": "1000000",
   "initialFailoverVersion": "the number you picked"
   "isGlobalNamespaceEnabled": true
   ```

4. Create a CA certificate as a PEM file and share it in your support ticket. This is required for [mutual TLS](https://www.cloudflare.com/learning/access-management/what-is-mutual-tls/) (mTLS)
   between your s2s-proxy (as client) and the Temporal s2s-proxy (as server). Note that this differs from the CA used for [accessing namespace](https://docs.temporal.io/cloud/certificates)
   in Temporal Cloud.

### Deploying the self-hosted s2s proxy

Complete the following steps in order to deploy and test the self-hosted s2s proxy. The proxy needs to be accessible from your server and requires a public internet connection to connect
to the Temporal s2s-proxy via migration endpoint.

> 1. Obtain the latest Docker image from the temporalio/s2s-proxy [repository](https://hub.docker.com/r/temporalio/s2s-proxy/tags).

Reviewer: ```suggestion
1. Obtain the latest Docker image from the [temporalio/s2s-proxy repository](https://hub.docker.com/r/temporalio/s2s-proxy/tags).
```

5. Request the Temporal team to perform a similar test to verify that the migration server can access your server.

Proceed only when you have recieved confirmation from Temporal that connectivity tests have passed.

### Migration process

Once all prerequisites have been completed, migration may commence.

#### Migration start

A Temporal operator will initiate the following migration commands on your behalf. The migration proceeds through these phases:

1. Start migration - Your self-hosted namespace remains active while the cloud namespace becomes passive. Workflows are replicated from the self-hosted namespace to the cloud namespace.
   Once cloud namespace is in-sync with self-hosted namespace, migration is ready for handover.
2. Handover to the Cloud - The cloud namespace becomes active and the self-hosted namespace becomes passive. Workflows are then replicated from the cloud namespace to the self-hosted server.
3. Confirm migration completion - Workflow replication from the cloud namespace to the self-hosted server stops.

> Before the final step, it is possible to handover the namespace back to the self-hosted server or abort the migration process completely. After this step, migrating namespaces from the cloud

Reviewer: ```suggestion
Before the final step, it is possible to hand over the namespace back to the self-hosted server or abort the migration process completely. After this step, migrating namespaces from the cloud
```

  continue to make progress during the handover, even if your cloud Temporal client is unable to access the cloud (due to misconfiguration, for example). The process is as follows:
  1. Direct your cloud-based Temporal clients to the cloud namespace endpoint. Initially, these clients will connect and send Poll requests but will not receive any tasks.
  2. Start migration. Your self-hosted namespace is active while your cloud namespace is passive (or standby). Your cloud Temporal clients can begin receiving tasks, but all requests
     from cloud clients to the Cloud namespace will automatically forward from the cloud to your self-hosted server.
  3. Hand over namespace to the cloud. Your cloud namespace becomes active and your self-hosted namespace becomes passive. All requests from your self-hosted Temporal clients will
     automatically forward from your server to the cloud.
  4. Complete migration. Your self-hosted Temporal clients will no longer receive any tasks from your server, allowing you to stop these clients.

> - Option 2 (not recommended)- Deploy one set of Temporal clients and switch the namespace endpoint during migration. With this option, your workflows will stop making progress if

Reviewer: ```suggestion
- Option 2 (not recommended)- Deploy one set of Temporal clients and switch the namespace endpoint during migration. With this option, your workflows will stop making progress.
```

>   this occurs. The process is as follows:

Reviewer: ```suggestion
  The process is as follows:
```
